src/plugins/discover/common/types.ts

```diff
@@ -6,19 +6,35 @@
  * Side Public License, v 1.
  */
 
-import type { ISearchSource } from '../../../../data/public';
-import { DiscoverGridSettingsColumn } from '../../components/discover_grid/types';
-import { VIEW_MODE } from '../../components/view_mode_toggle';
+import type { ISearchSource } from '../../data/common';
+import { SavedObjectAttributes } from '../../../core/types';
+import { VIEW_MODE } from './constants';
+
+export interface DiscoverGridSettingsColumn {
+  width?: number;
+}
+
+export interface DiscoverGridSavedSearch {
+  columns?: {
+    [key: string]: DiscoverGridSettingsColumn;
+  };
+}
+
+export interface SavedSearchGridAttributes extends SavedObjectAttributes {
+  columns: {
+    [key: string]: {
+      width?: number;
+    };
+  };
+}
 
 /** @internal **/
-export interface SavedSearchAttributes {
+export interface SavedSearchAttributes extends SavedObjectAttributes {
   title: string;
-  sort: Array<[string, string]>;
+  sort: string[];
```

Comment:
@pgayvallet changed it to satisfy `SavedObjectAttributes` , but it's wrong this way

Reply:
@pgayvallet should I provide more context, added a comment in `search_migration.ts`, here I took code from 

https://github.com/andrewctate/kibana/blob/3be8e9b96db16f96a73bc4de718dd073d271636d/src/plugins/visualizations/server/migrations/visualization_saved_object_migrations.ts#L1195

That's why I used `extends SavedObjectAttributes` to prevent a type error there

Reply:
Seems like our `SavedObjectAttributes` type is not currently accepting arrays of arrays of primitives, which are yet valid as attributes...

Due to TS's limitation on recursive types / type-loops, I'm not 100% sure how we can work around that

Reply:
I opened https://github.com/elastic/kibana/issues/128346. I would just work around with `@ts-expect-error` on the `sort` attribute definition until this get fixed

Reply:
thx! makes sense!